### PR TITLE
Trademark Search API 

### PR DESCRIPTION
**Pull request by HanilJain**
### **Solved** #998 

**Description**

Trademark search feature shows that trademark is available or not 
and If available shows the results fetched from the USPTO API 

**Setup Up**

https://rapidapi.com/pentium10/api/uspto-trademark

API_KEY can be derived from this URL and setted up in the env , 
If not the trademark feature wont work.

- /trademarks/
Screenshot 2024-02-24 032307

- /trademarks/query=youtube
Screenshot 2024-02-24 032342

- /trademarks/query=shirsh
Screenshot 2024-02-24 032533

### CODE REVIEW

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
This looks awesome. Can you please make the colors the same red as the logo for BLT and maybe not have any of the blue color just the red highlights also, can you please include the search term when it says 37 results please say 37 Results for searc.

**Comment by HanilJain** [comment]
> This looks awesome. Can you please make the colors the same red as the logo for BLT and maybe not have any of the blue color just the red highlights also, can you please include the search term when it says 37 results please say 37 Results for searc.

Done 👍 
Screenshot 2024-02-24 153437
Screenshot 2024-02-24 153451
Screenshot 2024-02-24 153514

**Comment by DonnieBLT** [comment]
Can you please run the pre-commit and make sure everything is formatted correctly

**Comment by HanilJain** [comment]
> Can you please run the pre-commit and make sure everything is formatted correctly

@DonnieBLT  Done 👍

File: blt/settings.py [hunk 1]
```diff
@@ -519,3 +519,7 @@
 APP_KEY_SECRET = os.environ.get("APP_KEY_SECRET")
 ACCESS_TOKEN = os.environ.get("ACCESS_TOKEN")
 ACCESS_TOKEN_SECRET = os.environ.get("ACCESS_TOKEN_SECRET")
+
+# UPSTO
```
Comment (arkid15r) [inline blt/settings.py:523]:
```suggestion
# USPTO
```

File: website/templates/trademark_detailview.html [hunk 2]
```diff
@@ -0,0 +1,155 @@
+{% extends "base.html" %}
+{% load gravatar %}
```
Comment (arkid15r) [inline website/templates/trademark_detailview.html:2]: Do you use these template tags?

**Review by arkid15r — COMMENTED** [review]
A couple of comments:

File: website/templates/trademark_detailview.html (hunk 2 above)
Comment (HanilJain) [reply website/templates/trademark_detailview.html]: Thanks I forget to look over them 
Removed in commit ab6e601efa5fca5e3af1f7a3ee3532c480c10a4d

**Comment by HanilJain** [comment]
> A couple of comments:

@arkid15r Thanks for the suggestion I missed few things.
are there any other changes required ?

File: website/templates/trademark_detailview.html [hunk 3]
```diff
@@ -0,0 +1,150 @@
+{% extends "base.html" %}
+{% load static %}
```
Comment (arkid15r) [inline website/templates/trademark_detailview.html:2]: Either use it at L80 or remove it.

File: website/templates/trademark_search.html [hunk 4]
```diff
@@ -0,0 +1,109 @@
+{% extends "base.html" %}
+{% load i18n %}
```
Comment (arkid15r) [inline website/templates/trademark_search.html:2]: I'm not sure if this one is in use as well.

Comment (HanilJain) [reply website/templates/trademark_search.html]: I don't why I forget this but now removed it 👍

File: website/templates/trademark_detailview.html (hunk 3 above)
Comment (HanilJain) [reply website/templates/trademark_detailview.html:2]: @arkid15r  Done 👍

**Review by arkid15r — COMMENTED** [review]
LGTM, great work!

**Comment by AtmegaBuzz** [comment]
Great work @HanilJain, one more thing how can we get the API key can you comment the specific API you used so that we can get that, also are there any limits to this API ?

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
One tweak. Also, can you maybe send us the key that you used or create a new key for us?

File: website/templates/trademark_search.html [hunk 5]
```diff
@@ -0,0 +1,108 @@
+{% extends "base.html" %}
+
+{% block title %}trademarks{% endblock %}
+
+{% block content %}
+{% include 'includes/sidenav.html' %}
+<head>
+    <link rel="preconnect" href="https://fonts.googleapis.com">
```
[the remark this answers is not in the document]
Comment (HanilJain) [reply website/templates/trademark_search.html:16]: @DonnieBLT changed in this PR #1858

**Comment by HanilJain** [comment]
Closing this PR due to  complicated merge conflicts 
For any future changes in this PR please refer #1858

**Pull request closed without merging**